collect_app/src/main/java/org/odk/collect/android/readytosend/ReadyToSendBanner.kt

```diff
@@ -0,0 +1,51 @@
+package org.odk.collect.android.readytosend
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import androidx.constraintlayout.widget.ConstraintLayout
+import org.odk.collect.android.databinding.ReadyToSendBannerBinding
+import org.odk.collect.forms.instances.Instance
+import org.odk.collect.forms.instances.InstancesRepository
+import org.odk.collect.strings.R
+import java.util.function.Supplier
+
+const val ONE_SECOND = 1000L
+const val ONE_MINUTE = 60000L
+const val ONE_HOUR = 3600000L
+const val ONE_DAY = 86400000L
+
+class ReadyToSendBanner(context: Context, attrs: AttributeSet?) : ConstraintLayout(context, attrs) {
+    constructor(context: Context) : this(context, null)
+
+    private val binding = ReadyToSendBannerBinding.inflate(LayoutInflater.from(context), this, true)
+
+    fun init(instancesRepository: InstancesRepository, clock: Supplier<Long>) {
+        val sentInstances: List<Instance> = instancesRepository.getAllByStatus(Instance.STATUS_SUBMITTED)
```

Comment:
We shouldn't really be accessing the DB like this from the UI thread. I think a better structure for this would be to have the sent instances and the number of instances ready to send load in a `ViewModel` (maybe a new `ReadyToSend` one) using `Scheduler#immediate` and expose them with `LiveData` from the same `ViewModel`.

Reply:
Done.